### ISSUE TITLE
Socket.IsDualMode getter throws exception

### ISSUE THREAD

**Issue by sipsorcery** [dotnet/runtime#33931]
The decision to throw an exception in the [System.Net.Sockets.Socket.IsDualMode](https://github.com/dotnet/runtime/blob/963b1587b48ec2fa09cb5bcd0a53460762a459e4/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs#L631) seems strange. Why not just return false?

As it stands any call to `IsDualMode` needs to have a pre-check for the `AddressFamily == AddressFamily.InterNetworkV6` to avoid the exception.

**Comment by karelz** [comment]
Triage: We agree the property should return `false` instead of throwing. `GetOption` should throw.
We may need update of docs and likely document it as technical breaking change.

@sipsorcery are you interested in creating a PR?

**Comment by sipsorcery** [comment]
I agree the workaround is straight forward... once you know about it. My guess is they'll be a steady stream of devs hitting this exception and needing to dig into the docs/source to find out what's going on. Seems a shame to put that overhead on them for no discernible benefit?

And not to be pedantic but the MS library design guidelines do say to [avoid exceptions in property getters](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property).

**Comment by karelz** [comment]
Good point about design guidelines.
The API has been around since .NET Framework 4.5, behaving this way: https://apisof.net/catalog/System.Net.Sockets.Socket.DualMode ... so I am not sure we will see suddenly more devs hitting it.
